ethereum/datastructures/src/test/java/tech/pegasys/artemis/datastructures/operations/SlashableAttestationTest.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.datastructures.operations;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static tech.pegasys.artemis.datastructures.util.DataStructureUtil.randomAttestationData;
+import static tech.pegasys.artemis.datastructures.util.DataStructureUtil.randomUnsignedLong;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import net.consensys.cava.bytes.Bytes;
+import net.consensys.cava.bytes.Bytes32;
+import net.consensys.cava.bytes.Bytes48;
+import org.junit.jupiter.api.Test;
+
+class SlashableAttestationTest {
+
+  private List<UnsignedLong> validatorIndices =
+      Arrays.asList(randomUnsignedLong(), randomUnsignedLong(), randomUnsignedLong());
+  private AttestationData data = randomAttestationData();
+  private Bytes32 custodyBitfield = Bytes32.random();
+  private BLSSignature aggregateSignature = new BLSSignature(Bytes48.random(), Bytes48.random());
+
+  private SlashableAttestation slashableAttestation =
+      new SlashableAttestation(validatorIndices, data, custodyBitfield, aggregateSignature);
+
+  @Test
+  void equalsReturnsTrueWhenObjectAreSame() {
+    SlashableAttestation testSlashableAttestation = slashableAttestation;
+
+    assertEquals(slashableAttestation, testSlashableAttestation);
+  }
+
+  @Test
+  void equalsReturnsTrueWhenObjectFieldsAreEqual() {
+    SlashableAttestation testSlashableAttestation =
+        new SlashableAttestation(validatorIndices, data, custodyBitfield, aggregateSignature);
+
+    assertEquals(slashableAttestation, testSlashableAttestation);
+  }
+
+  @Test
+  void equalsReturnsFalseWhenValidatorIndicesAreDifferent() {
+    // Create copy of validatorIndices and reverse to ensure it is different.
+    List<UnsignedLong> reverseValidatorIndices = new ArrayList<>(validatorIndices);
+    Collections.reverse(reverseValidatorIndices);
+
+    SlashableAttestation testSlashableAttestation =
+        new SlashableAttestation(
+            reverseValidatorIndices, data, custodyBitfield, aggregateSignature);
+
+    assertNotEquals(slashableAttestation, testSlashableAttestation);
+  }
+
+  @Test
+  void equalsReturnsFalseWhenCustodyBitfieldIsDifferent() {
+    Bytes32 otherCustodyBitfield = custodyBitfield.and(Bytes32.random());
+
+    SlashableAttestation testSlashableAttestation =
+        new SlashableAttestation(validatorIndices, data, otherCustodyBitfield, aggregateSignature);
```

Comment:
Since `custodyBitfield` is just a Bytes32, instead of generating a new random Bytes32, just using `custodyBitfield.not()` should also ensure that the type under test is different.